src/Protocols/PacketStream/Base.hs

```diff
@@ -0,0 +1,213 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+{- |
+Definitions and instances of the packet stream protocol
+-}
+module Protocols.PacketStream.Base (
+  PacketStreamM2S (..),
+  PacketStreamS2M (..),
+  PacketStream,
+  unsafeToPacketStream,
+  fromPacketStream,
+  forceResetSanity,
+  filterMetaS,
+  filterMeta,
+  mapMetaS,
+  mapMeta,
+) where
+
+import Clash.Prelude hiding (sample)
+import qualified Prelude as P
+
+import qualified Protocols.Df as Df
+import qualified Protocols.DfConv as DfConv
+import Protocols.Hedgehog.Internal
+import Protocols.Internal
+
+import Control.DeepSeq (NFData)
+import Data.Coerce (coerce)
+import qualified Data.Maybe as Maybe
+import Data.Proxy
+
+{- | Data sent from manager to subordinate, a simplified AXI4-Stream like interface
+with metadata that can only change on packet delineation.
+_tdest, _tuser and _tid are bundled into one big _meta field which holds metadata.
+There are no null or position bytes so _tstrb is replaced by a last indicator
+that indicates the index of the last valid byte in the _data vector.
+_tvalid is modeled via wrapping this in a `Maybe`.
```

Comment:
A `_tvalid` is needed in case the manager is released from its reset later than the subordinate it is talking to. Multiple resets is perfectly fine from a metastability point of view when you have synchronous resets. And when you have components which you think do handshaking, then it's not too weird to have different reset deassert times to achieve timing closure.

Reply:
I thought the “wrapping” in a Maybe was optional. But I now see it is part of the Protocol instance. Apologies for the misunderstanding.